services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/multipart/KnownContentLengthAsyncRequestBodySubscriber.java

```diff
@@ -138,6 +144,7 @@ public void onNext(AsyncRequestBody asyncRequestBody) {
             partNumber.getAndIncrement();
             asyncRequestBody.subscribe(new CancelledSubscriber<>());
             subscription.request(1);
+            updateProgress(asyncRequestBody.contentLength().orElse(0L));
```

Comment:
Reporting progress as 0 if content-length is not present seems a bit odd, should we just not reporting it if that's the case?